crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -1081,7 +1081,7 @@ mod test {
     // Helper function to generate a vector of bits for testing purposes
     fn generate_be_bits(x: &Felt) -> Vec<bool> {
         // Initialize an empty vector to store the expected bits
-        let mut bits = Vec::new();
+        let mut bits = Vec::with_capacity(x.0.representative().limbs.len() * 8 * 8);
```

Comment:
The call to representative is not free at all.
Now there are two of those.

If you want to do this you should 

```
let representative = x.0.representative();

let mut bits = Vec::with_capacity(limbs.len() * 8 * 8);

for limb in representative.limbs {
```

Reply:
This is not what you did.